src/main/java/seedu/address/logic/parser/ClientNoteAddCommandParser.java

```diff
@@ -25,20 +27,28 @@
     public ClientNoteAddCommand parse(String userInput) throws ParseException {
         requireNonNull(userInput);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(userInput, /* PREFIX_TAG,*/ PREFIX_NOTE);
+                ArgumentTokenizer.tokenize(userInput, PREFIX_TAG, PREFIX_NOTE);
         if (!arePrefixesPresent(argMultimap, PREFIX_NOTE) || argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ClientNoteAddCommand.MESSAGE_USAGE));
         }
         Index index;
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            String[] preambleWords = argMultimap.getPreamble().split(" "); // todo: ask if there's a better way?
+            index = ParserUtil.parseIndex(preambleWords[preambleWords.length - 1]);
         } catch (ParseException pe) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                     ClientNoteAddCommand.MESSAGE_USAGE), pe);
         }
+        Tag tag;
+        try {
+            tag = ParserUtil.parseTag(argMultimap.getValue(PREFIX_TAG).orElse("untagged"));
```

Comment:
Might want to consider extracting `"untagged"` into a constant variable.

Reply:
> 
> 
> > Quite a few TODOs and stray comments. I think a follow-up issue can be created to cleanup the todos and stray comments.
> > Just wondering, what is the main purpose of having `TagNoteMap` class? Is it so that it would be easier to query for notes based on a given set of Tags? Because it seems like additional work to "sync" tags with notes and it might be quite error-prone if the syncing is not done right. Any thoughts @qwoprocks ?
> 
> @raysonkoh actually the implementation @rtshkmr did was a bit off from what I had in mind, the tagnotemap is actually supposed to be the one being passed down from logic manager to the parser to create the tags, so there would no need to do the syncing stuff. @rtshkmr isit alright if i just push to this PR and implement that?


let me clean it up based on both your comments first